### PR TITLE
Add missing HTTP status codes/messages

### PR DESCRIPTION
**Pull request by ravage84**
Add status codes from [Zend/Diactoros](https://github.com/zendframework/zend-diactoros/blob/215f47a73fb2a7f2d33230a3e23658acae234035/src/Response.php#L32-L96) and [httpstatuses.com](https://httpstatuses.com/).

Refers https://github.com/cakephp/cakephp/issues/9516#issuecomment-249367130

### CODE REVIEW

File: src/Network/Response.php [hunk 1]
```diff
@@ -71,15 +77,30 @@ class Response
...
         502 => 'Bad Gateway',
         503 => 'Service Unavailable',
         504 => 'Gateway Time-out',
         505 => 'Unsupported Version'
+        506 => 'Variant Also Negotiates',
+        507 => 'Insufficient Storage',
+        508 => 'Loop Detected',
+        510 => 'Not Extended',
+        511 => 'Network Authentication Required',
+        511 => 'Network Connect Timeout Error',
```
Comment (chinpei215) [inline src/Network/Response.php]: 599?

Comment (ravage84) [inline src/Network/Response.php]: Darn! Copy-Pasta Error :imp:

File: src/Network/Response.php [hunk 2]
```diff
@@ -71,15 +77,30 @@ class Response
...
+        423 => 'Locked',
+        424 => 'Failed Dependency',
+        425 => 'Unordered Collection',
+        426 => 'Upgrade Required',
+        428 => 'Precondition Required',
         429 => 'Too Many Requests',
+        431 => 'Request Header Fields Too Large',
+        444 => 'Connection Closed Without Response',
         451 => 'Unavailable For Legal Reasons',
+        499 => 'Client Closed Requet',
```
Comment (inoas) [inline src/Network/Response.php]: typo

File: src/Network/Response.php [hunk 3]
```diff
@@ -71,15 +77,30 @@ class Response
...
         500 => 'Internal Server Error',
         501 => 'Not Implemented',
         502 => 'Bad Gateway',
         503 => 'Service Unavailable',
         504 => 'Gateway Time-out',
```
Comment (inoas) [inline src/Network/Response.php]: Timeout

Comment (ravage84) [inline src/Network/Response.php]: I did not want to change the existing ones. Can be done in another PR...

File: src/Network/Response.php [hunk 4]
```diff
@@ -71,15 +77,30 @@ class Response
...
+        444 => 'Connection Closed Without Response',
         451 => 'Unavailable For Legal Reasons',
+        499 => 'Client Closed Request',
         500 => 'Internal Server Error',
         501 => 'Not Implemented',
         502 => 'Bad Gateway',
         503 => 'Service Unavailable',
         504 => 'Gateway Time-out',
         505 => 'Unsupported Version'
+        506 => 'Variant Also Negotiates',
```
Comment (markstory) [inline src/Network/Response.php]: Missing a comma on line 97.

**Review by markstory — CHANGES_REQUESTED** [review]

**Comment by ADmad** [comment]
Failing test seems relevant to this change.

**Comment by ADmad** [comment]
I'll fix the test case post merge.

**Comment by ravage84** [comment]
@ADmad you are too fast, just pushed the updated test code ;-)

**Comment by inoas** [comment]
306 is wrong. I fixed it here: https://github.com/cakephp/cakephp/pull/9528
Also I carried together 65 codes. One is missing in this PR.

**Comment by inoas** [comment]
`421 => 'Misdirected Request', // RFC7540, Section 9.1.2` is missing in this PR, I fixed it here: #9528

**Pull request merged**